CMakeLists.txt

```diff
@@ -299,7 +306,9 @@ else ()
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_LOGICAL_OP -Wlogical-op)
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_RANGE_LOOP_ANALYSIS -Wrange-loop-analysis)
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_TAUTOLOGICAL_ZERO_COMPARE -Wtautological-unsigned-zero-compare)
-    ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_OLD_STYLE_CAST -Wold-style-cast)
+    if (NOT APPLE)
+        ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_OLD_STYLE_CAST -Wold-style-cast)
```

Comment:
Why this change? We shouldn't need to use old style casts on any platform

Reply:
I got this because of openssl and its old C cast style from its headers :-)

Reply:
Then we need to make sure SSL is identified as a system dependency and not the actual code

Reply:
@devnexen can you address that, please?

Reply:
Thing is that we don't want to disable this warning. We need to find out where in the MacOS build the openSSL and other library are being included and tell the build system that it is a system include